File: segundo/1er cuatri/PDOO/irrgarten_entrega1/Irrgarten_java/src/irrgarten/TestP2.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author alvarogz
 */
public class TestP2 {
    public static void main(String[] args) {
        System.out.println("----Monster-----");
        Monster m = new Monster ("m", Dice.randomIntelligence(), Dice.randomStrength());
        m.setPos(1, 2);
        System.out.println(m.toString());
        
        if (m.dead())
            System.out.println("Esta muerto");
        else
            System.out.println("NO esta muerto");
        
        System.out.println(m.attack());
        
        System.out.println("---------------------");
       
      
        System.out.println("----Player-----");
        Player p = new Player ('1', Dice.randomIntelligence(), Dice.randomStrength());
        System.out.println(p.toString());
        System.out.println(p.attack());
   
        if (p.dead())
            System.out.println("Esta muerto");
        else
            System.out.println("NO esta muerto");

       
        while (p.dead()) p.resurrect();
        
        System.out.println(p.toString());
        
        System.out.println("----------------------------");
      
        System.out.println("---------Labyrinth----------");
        Labyrinth l = new Labyrinth (3, 3, 2, 2);
        System.out.println(l.toString());
        System.out.println("Añado el monstruo m :");
        l.addMonster(1, 1, m);   
       
        System.out.println(l.toString());
        int pos[];
        for(int i = 0; i < 20; i++){
            pos= l.randomEmptyPos();
            System.out.println("\n pos: "+pos[0] +", "+ pos[1] +"\n");
            
        }
        
        System.out.println("----------------------------");
        
        System.out.println("------------Game------------");
        
        System.out.println("----------------------------");
    }
}

File: segundo/1er cuatri/PDOO/irrgarten_entrega1/irrgarten_ruby/Irrgarten/TestP2.rb
# frozen_string_literal: true
require_relative 'monster'
require_relative 'dice'
require_relative 'player'
require_relative 'labyrinth'

class TestP2
  puts "----------Monster-----------"
  m = Monster.new("p", 2, 5)

  m.set_pos(1,2)
  if m.dead == true
    puts "Estoy muerto"
  else
    puts "Estoy vivo"
  end

  puts m.attack
  puts m.to_s
  m.got_wounded
  puts m.to_s


  puts "----------Player-----------"
  p = Player.new("1", Dice.random_intelligence, Dice.random_strength)
  puts p.to_s
  puts p.attack
  if p.dead == true
    puts "Estoy muerto"
  else
    puts "Estoy vivo"
  end

  begin
    p.resurrect
  end while p.dead == true

  puts p.to_s

  puts "----------Labyrinth-----------"
  l = Labyrinth.new(3,3,2,2)
  puts l.to_s
  puts "Añado el monstruo m :"
  l.add_monster(1,1,m)
  puts l.to_s
  pos = Array.new

  for i in 0...20 do
    pos = l.random_empty_pos
    puts "\n pos: #{pos[0]}, #{pos[1]} \n"
  end
end